### DISCUSSION TITLE
Can we run a webserver alongside Il2Cpp.perform?

### DISCUSSION THREAD

**Discussion by TheMegafuji** [vfsfitvnm/frida-il2cpp-bridge#348] [Q&A]
I've been trying to import different libs to have some sort of customization on the running code, I know its more supposed to be like a debugging manner, but thinking of a final product somehow, I wanted to be able to have more flexibility on customizing the running code with some interface.
I've tried communicating with web sockets and http requests, but didn't find a way still, also I was wondering if there could be a all in one solution, as running an express server in the same script that is built with frida-compile, but since its a ECMAscript I think that is quite a hard task.

Thinking of a solution that would have some user inputs in the running code/instances of Il2Cpp methods, classes, static or not, how one could build a GUI or webserver to work with it?

**Comment by vfsfitvnm** [comment]
(relevant discussion https://github.com/vfsfitvnm/frida-il2cpp-bridge/issues/320)

Yes, you *could* do that - keep in mind the target app must have permissions to make network requests. 
Moreover, you'd have to find a way to embed such webserver: `frida-compile` has shims ([list](https://github.com/frida/frida-compile/blob/3a66562614ba06140e8e9011e96e33122e42d7ff/src/compiler.ts#L257)) for certain packages (including `https`), so you could try with that.

**Comment by 350030173** [comment]
https://github.com/GrowthEase/GameSentry
